third_party/txt/src/txt/font_skia.cc

```diff
@@ -24,21 +25,14 @@ namespace {
 
 hb_blob_t* GetTable(hb_face_t* face, hb_tag_t tag, void* context) {
   SkTypeface* typeface = reinterpret_cast<SkTypeface*>(context);
-
-  const size_t table_size = typeface->getTableSize(tag);
-  if (table_size == 0)
-    return nullptr;
-  void* buffer = malloc(table_size);
-  if (buffer == nullptr)
-    return nullptr;
-
-  size_t actual_size = typeface->getTableData(tag, 0, table_size, buffer);
-  if (table_size != actual_size) {
-    free(buffer);
+  sk_sp<SkData> data = typeface->copyTableData(tag);
+  if (!data) {
     return nullptr;
   }
-  return hb_blob_create(reinterpret_cast<char*>(buffer), table_size,
-                        HB_MEMORY_MODE_WRITABLE, buffer, free);
+  SkData* rawData = data.release();
+  return hb_blob_create(reinterpret_cast<char*>(rawData->writable_data()),
```

Comment:
Harfbuzz shouldn't need to modify the table data. In that case, `HB_MEMORY_MODE_READONLY` is more appropriate. I see this specification was made in the old code. We should update to qualify access mode.

Reply:
HarfBuzz sometimes need to modify the data. `HB_MEMORY_MODE_WRITABLE` mean the table data are solely for HarfBuzz use, otherwise it will make an internal copy of the data.